pkg/reconciler/configuration/configuration.go

```diff
@@ -145,33 +145,35 @@ func (c *Reconciler) findAndSetLatestReadyRevision(config *v1.Configuration) err
 
 // getSortedCreatedRevisions returns the list of created revisions sorted in descending
 // generation order between the generation of the latest ready revision and config's generation (both inclusive).
-func (c *Reconciler) getSortedCreatedRevisions(config *v1.Configuration) ([]*v1.Revision, error) {
+func (c *Reconciler) getSortedCreatedRevisions(ctx context.Context, config *v1.Configuration) ([]*v1.Revision, error) {
+	logger := logging.FromContext(ctx)
 	lister := c.revisionLister.Revisions(config.Namespace)
 	configSelector := labels.SelectorFromSet(labels.Set{
 		serving.ConfigurationLabelKey: config.Name,
 	})
 	if config.Status.LatestReadyRevisionName != "" {
 		lrr, err := lister.Get(config.Status.LatestReadyRevisionName)
+		// Record the error and continue because we still want to set the LRR to the correct revision.
 		if err != nil {
-			return nil, err
-		}
-		start := lrr.Generation
-		var generations []string
-		for i := start; i <= int64(config.Generation); i++ {
-			generations = append(generations, strconv.FormatInt(i, 10))
-		}
+			logger.Errorf("Error getting latest ready revision %q: %v", config.Status.LatestReadyRevisionName, err)
```

Comment:
Can we add a detailed comment here?

Something like:
```suggestion
			// If the user deletes the LatestReadyRevision then this may return an error due to the
			// dangling reference.  Proceed to calculate the next-latest ready revision so that the
			// caller can synthesize a new Revision at the current generation to replace the one deleted.
			logger.Errorf("Error getting latest ready revision %q: %v", config.Status.LatestReadyRevisionName, err)
```